New Version History guides for Dashboards and Notebooks (#18207)
DataDog/documentation: content/en/notebooks/guide/version_history.md

---
title: Version History for Notebooks
kind: documentation
disable_toc: false
further_reading:
- link: "/notebooks/"
  tag: "documentation"
  text: "Notebooks Overview"
- link: "/account_management/audit_trail/"
  tag: "documentation"
  text: "Audit Trail Overview"
---

## Overview
Version History automatically tracks changes made to your Notebooks and saves previous versions so you can see exactly what was changed and by whom. With this feature you can view previous versions, restore your notebook to any saved version, or clone a version to create a new notebook.

## Prerequisites
All notebooks have a default 30 days retention of version history. In order to see any previous versions, an edit must be made within the last 30 days. 

With [Audit Trail][1] enabled, the version history is extended from 30 days to 90 days. After enabling Audit Trail, you are able to see any edits made between 30 to 90 days ago on all existing notebooks. 

## View versions

> From a notebook click on the cog icon in the top right corner of the page and select the **Version History** option in the menu to open the Version History side panel. If there are no edits within the retention period, the Version History option is disabled.

Reviewer: ```suggestion
From a notebook, click the **Configure** icon and click **Version History** to open the Version History side panel. If there are no edits within the retention period, the Version History option is disabled.
```

{{< img src="/notebooks/guide/version_history/disabled_version_history.png" alt="Disabled version history option in a notebook" style="width:100%;" >}}

In the Version History side panel, for each version you can see:
- Which Datadog user made the change
- The date and time of the change
- A summary of the change and a detailed change description of the version to its predecessor

## Preview a version
From the Version History side panel, click on a version to preview what your notebook would look like if you choose to restore to that version. Click on any version to scroll to the location of the change and highlight any widgets or cells that were changed.

**Note**: Clicking on a version to preview it does not save any changes or impact what other users see until you actively choose to restore to that version.

## Restore a version
There are two ways you can restore your notebook to a previous version.

{{< img src="/notebooks/guide/version_history/version_history_options.png" alt="Example notebooks showing version history options" style="width:100%;" >}}

> - From the Version History side panel, after you choose the version to restore, click the three dot ellipsis menu and select **Restore this version**.

Reviewer: ```suggestion
- From the Version History side panel, after you choose the version to restore, click the kebab menu to the right of a user profile and select **Restore this version**.
```
